### PR TITLE
gm: Use default values when `gm.toml` is missing or incomplete

### PR DESCRIPTION
**Pull request by greg-szabo**
Closes: #993 

## Description

* default values apply better: you can have a minimal configuration and details will be filled in
* Added a `global.hermes` section to separate hermes config better
* bumped version to 0.0.3
* updated gm changelog
* updated stoml dependency to 0.7.0
______

For contributor use:

- [X] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [X] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Review by adizere — CHANGES_REQUESTED** [review]
Could you please also update the example `gm.toml` in https://github.com/informalsystems/ibc-rs/blob/master/scripts/gm/gm.toml ?

**Comment by greg-szabo** [comment]
I'm not sure what you mean, the last file in the file explorer is the gm.toml example. You added the link from master, which is going to be updated as soon as this PR is merged.
Here is the file from this PR: https://github.com/informalsystems/ibc-rs/blob/4baec49fd92bd775614a201b2fc5fc00bcb701c1/scripts/gm/gm.toml

File: scripts/gm/gm.toml [hunk 1]
```diff
@@ -26,11 +26,20 @@ auto_maintain_config=true
...
-# (Optional) Hermes configuration file path. Ignored in sub-sections.
-hermes_config="$HOME/.hermes/config.toml"
+# (Optional) Path to the `hermes` binary to use.
+binary="./hermes"
+
+# (Optional) Hermes configuration file path.
+config="$HOME/.hermes/config.toml"
+
+# Hermes configuration strategy paremeter.
+strategy="naive"
```
Comment (adizere) [inline scripts/gm/gm.toml:39]: Let's put this "all" so that it's consistent with the

**Comment by greg-szabo** [comment]
I've updated the default strategy to "packets" in the code too.

**Comment by adizere** [comment]
> I'm not sure what you mean, the last file in the file explorer is the gm.toml example. You added the link from master, which is going to be updated as soon as this PR is merged.
> Here is the file from this PR: https://github.com/informalsystems/ibc-rs/blob/4baec49fd92bd775614a201b2fc5fc00bcb701c1/scripts/gm/gm.toml

My bad. I somehow missed the [.toml](https://github.com/informalsystems/ibc-rs/blob/4baec49fd92bd775614a201b2fc5fc00bcb701c1/scripts/gm/gm.toml) when browsing in the file explorer. 

Thanks for the quick response!

There is one more hiccup I'm having (which may be due to some misunderstanding, not `gm`), specifically after I set up everything with `gm start; gm hermes config; gm hermes keys`, I do the following:

```
hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 9999 -o 1000
Error: tx error: failed with underlying cause: deliver_tx reports error: log=Log("failed to execute message; message index: 0: 0samoleans is smaller than 1samoleans: insufficient funds")
```

From what I can tell we're loading both stake and samoleans, but only transfers of stake works:

```
hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 1 -o 1000 -d stake
    Finished dev [unoptimized + debuginfo] target(s) in 0.92s
     Running `target/debug/hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 1 -o 1000 -d stake`
Jun 01 18:22:11.826  INFO ibc_relayer_cli::commands: using default configuration from '/Users/adi/.hermes/config.toml'
Success: [
    SendPacket(
        SendPacket {
            height: Height {
                revision: 0,
                height: 189,
            },
            packet: PortId("transfer") ChannelId("channel-0") Sequence(1),
        },
    ),
]
```

Any thoughts?

**Comment by greg-szabo** [comment]
The issue was a typo and I just fixed it in the latest commit. Make sure you regenerate your networks to test. (`gm rm; gm start`)

The issue was that I added `samoleans` contrary to `samolean`.

Previously this worked:
```
macbook15:ibc-rs greg$ hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 9999 -o 1000 -d samoleans
Jun 01 11:33:04.923  INFO ibc_relayer_cli::commands: Using default configuration from: '.hermes/config.toml'
Success: [
    SendPacket(
        SendPacket {
            height: Height {
                revision: 0,
                height: 2987,
            },
            packet: PortId("transfer") ChannelId("channel-0") Sequence(4),
        },
    ),
]```

**Comment by greg-szabo** [comment]
Huh, now it works...

I'm going to start from scratch and try to reproduce the issue. The only time I was able to do that was when the denomination wasn't available.

**Comment by greg-szabo** [comment]
Making notes of the steps:
1. Delete all old network config
2. Delete `.hermes/` folder
3. gm start
4. `gm hermes config`
5. `gm hermes keys`
6. Execute the result of `gm hermes cc`
7. Run `hermes tx raw ft-transfer ibc-1 ibc-0 transfer channel-0 9999 -o 1000`
Result:
```
Jun 01 13:59:09.388  INFO ibc_relayer_cli::commands: Using default configuration from: '.hermes/config.toml'
Success: [
    SendPacket(
        SendPacket {
            height: Height {
                revision: 0,
                height: 29,
            },
            packet: PortId("transfer") ChannelId("channel-0") Sequence(1),
        },
    ),
]
```

gm.toml:
```
[global]
  add_to_hermes = false
  gaiad_binary = "/Users/greg/go/bin/gaiad_4.2.1"
  ports_start_at = 27040

  [global.hermes]
    binary = "/Users/greg/go/bin/hermes_0.3.2"
    strategy = "naive"

[ibc-0]
  ports_start_at = 27000

[ibc-1]
  ports_start_at = 27010

[node-0]
  add_to_hermes = true
  network = "ibc-0"
  ports_start_at = 27020

[node1]
  add_to_hermes = true
  network = "ibc-1"
  ports_start_at = 27030
```

The transaction used up 9999 samoleans and 1000 stake (for the fee). So, although I thought I can reproduce the issue, now I can't. Can you try with a fresh config?

**Review by adizere — APPROVED** [review]
Worked like a charm with [the latest instructions](https://github.com/informalsystems/ibc-rs/pull/1014#issuecomment-852335035). Was not able to reproduce my problem with `samoleans` anymore (the salient difference is that I cleaned-up the `.hermes/` folder, but unclear how/if this fixed it).

Great stuff Greg!!

**Pull request merged**